tmt/steps/provision/vagrant.py

```diff
@@ -7,101 +7,267 @@
 import os
 import re
 import shutil
+import click
+#import pprint
 from time import sleep
 
-from tmt.steps.provision.base import ProvisionBase
+#from tmt.steps.provision.base import ProvisionBase
 from tmt.utils import ConvertError, SpecificationError, GeneralError, quote
 
-from click import echo
 from urllib.parse import urlparse
 
 
-# DATA[*]:
-#   HOW = libvirt|virtual|docker|container|vagrant|...
-#         provider, in Vagrant's terminilogy
-#
-#   IMAGE = URI|NAME
-#         NAME is for Vagrant or other HOW, passed directly
-#         URI can be path to BOX, QCOW2 or Vagrantfile f.e.
-#
-#   BOX = Set a BOX name directly (in case of URI for IMAGE)
-#
-class ProvisionVagrant(ProvisionBase):
-    """ Use Vagrant to Provision an environment for testing """
+class ProvisionVagrant(tmt.steps.provision.ProvisionPlugin):
+    """
+    Use Vagrant to Provision an environment for testing
+    """
+
+    # Guest instance
+    _guest = None
+
+    # Supported methods
+    _methods = [
+        tmt.steps.Method(name='virtual.vagrant', doc=__doc__, order=50),
```

Comment:
Does this priority (order) need to be unique, which one will be chosen. And if explicitly changed, should it be higher (60), or lower (40)?

Reply:
Yes, priority should be unique. Based on it the default plugin is used. Use `60` for vagrant.

Reply:
Ok, great!